src/storage.c

```diff
@@ -102,6 +103,34 @@ static int save_settings(int fd, struct l_settings *settings)
 	return err;
 }
 
+static int check_valid_groups(char **all_groups, int index)
+{
+	int repeated_check;
+	regex_t preg;
+	int i;
+	int err;
+
+	regcomp(&preg, "^DataItem_[0-9]+$", REG_EXTENDED);
+
+	if (!regexec(&preg, all_groups[index], 0, (regmatch_t *)NULL, 0) == 0) {
+		l_error("Invalid DataItem group: %s", all_groups[index]);
+		err = -EINVAL;
+	}
+
+	for (i = index + 1; all_groups[i] != NULL; i++) {
+		repeated_check = strcmp(all_groups[i], all_groups[index]);
+		if (repeated_check == 0) {
+			l_error("Repeated DataItem group: %s", all_groups[i]);
+			err = -EINVAL;
```

Comment:
you can break this for here if an err already occured

Reply:
Done!